Config overhaul including "setField" and better help functionality + getStatus

TheRedDaemon opened the pull request:
- The general plan can be read [HERE](https://github.com/gynt/stronghold-mapeditortools/issues/7#issuecomment-826112173) under 
  > Not a quote: Help texts, value assign, etc. Because it plays into each other.
- Bloats the file even more. However, formatting and other optimizations should be tackled latter.

TheRedDaemon committed:
[CONFIG] Created default config constructors

TheRedDaemon committed:
[CONFIG] Created base structure and first texts

- this approach bloats the code even more
- but it allows the control functions like intended

TheRedDaemon committed:
Merge branch 'main' of https://github.com/gynt/stronghold-mapeditortools into ConfigOverhaul

TheRedDaemon committed:
[CONFIG] Created set field and help structure

- help texts still missing

TheRedDaemon committed:
[CONFIG] created getStatus() function

- returns relevant fields
- help texts still not done

TheRedDaemon committed:
[CONFIG] Finished current help texts.

- mirror "point" is now default active
. renamed mirror functions

TheRedDaemon commented:
Plan is implemented. Also added the new status functionality.
Code is more messy than ever 🙃, but I would postpone a clean-up until the other features I originally prepared are also included.

gynt commented on shc-mapeditortools.lua:
```diff
@@ -765,61 +753,644 @@ end
+    fields = fields or {}
+    setmetatable(fields, self)
+    self.__index = self
+    self.__call = DefaultBase.setField -- sets __call always to default handler
```
self.__newindex = function(self, i, v)
      if self[i] == nil then
        print("parameter '"..i.."' does not exist")
      else
        rawset(self, i, v)
      end
    end

gynt replied:
It will do for now. We might consider making any assignment `config.key = value` trigger setField (when we do we change this snippet again), but then all assignment statements `config.key = value` in the set functions need to become `rawset` or we would get in an infinite loop :)

TheRedDaemon replied:
Created an issue for this question: [HERE](https://github.com/gynt/stronghold-mapeditortools/issues/9)
Also added the suggested guard to at least prevent invalid fields.

gynt commented:
I made one comment about __newindex. The snippet will make sure invalid assignments to a config are reported as an error.

TheRedDaemon committed:
[CONFIG] Added simple assign guard, preventing unknown fields

- like suggested
- directly assigning non-sense to valid fields is still possible
